Lib9c/Action/RewardGold.cs

```diff
@@ -283,27 +282,19 @@ public IWorld ResetChallengeCount(IActionContext ctx, IWorld states)
 
         public IWorld MinerReward(IActionContext ctx, IWorld states)
         {
-            // 마이닝 보상
-            // https://www.notion.so/planetarium/Mining-Reward-b7024ef463c24ebca40a2623027d497d
-            // Currency currency = states.GetGoldCurrency();
             Currency currency = Currencies.Mead;
-            FungibleAssetValue defaultMiningReward = currency * 10;
-            var countOfHalfLife = (int)Math.Pow(2, Convert.ToInt64((ctx.BlockIndex - 1) / 12614400));
-            FungibleAssetValue miningReward =
-                defaultMiningReward.DivRem(countOfHalfLife, out FungibleAssetValue _);
-
-            // var balance = states.GetBalance(GoldCurrencyState.Address, currency);
-            // if (miningReward >= FungibleAssetValue.Parse(currency, "1.25") && balance >= miningReward)
-            // {
-            //     states = states.TransferAsset(
-            //         ctx,
-            //         GoldCurrencyState.Address,
-            //         Addresses.RewardPool,
-            //         miningReward
-            //     );
-            // }
-
-            states = states.MintAsset(ctx, Addresses.RewardPool, miningReward);
+            var usedGas = states.GetBalance(Addresses.GasPool, currency);
+            var defaultReward = currency * 5;
+            var halfOfUsedGas = usedGas.DivRem(2).Quotient;
+            var gasToBurn = usedGas - halfOfUsedGas;
+            var miningReward = halfOfUsedGas + defaultReward;
+            states = states.MintAsset(ctx, Addresses.GasPool, defaultReward);
+            if (gasToBurn.Sign > 0)
+            {
+                states = states.BurnAsset(ctx, Addresses.GasPool, gasToBurn);
+            }
```

Comment:
Above code seems much clear though, handling defaultReward - gasToBurn seems more effective.
But since `RewardGold()` is called once per block, above seems better for here.

Reply:
> Above code seems much clear though, handling defaultReward - gasToBurn seems more effective. But since `RewardGold()` is called once per block, above seems better for here.

There's no need to fix the code, right?